File: Python/Desafio - aula 13.py
def trataValores():
    c = 1
    quantidade = 0
    numeros = []
    numerosPar = []
    numerosImpar = []
    
    try:
        quantidade = int(input("Digite quantos números você deseja inserir: "))
    except ValueError:
        print("PRECISA SER UM NUMERO INTEIRO!")

    while(c <= quantidade):
        try:
            n = float(input(f"Digite o número {c}: "))
            numeros.append(n)
            c+=1
        except ValueError:
            print("PRECISA SER UM NÚMERO!")

    print(f"A lista ficou assim {numeros}")

    for n in numeros:
        if n % 2 == 0:
            numerosPar.append(n)
        else:
            numerosImpar.append(n)

    print(f"Pares: {numerosPar}")
    print(f"Ímpares: {numerosImpar}")

    print(f"Soma pares: {sum(numerosPar)}")
    print(f"Soma ímpares: {sum(numerosImpar)}")

    print(f"Média: {sum(numeros)/quantidade}")

trataValores()

File: R/Desafio - aula 13.R
trataValores <- function(){
    c <- 1
    quantidade <- 1
    numeros <- c()
    numerosPar <- c()
    numerosImpar <- c()
    
    tryCatch({
        quantidade <- as.integer(readline(prompt = "Digite um valor: ")) 
    }, error={
        print("PRECISA SER UM NUMERO INTEIRO!")
    })

    while(c <= quantidade){
        tryCatch({
            n <- as.double(readline(prompt = paste("Digite o número", c, ": ")))
            append(numeros, n)
            c <- c+1
        },error={
            print("PRECISA SER UM NÚMERO!")
        })
    }

    print(paste("A lista ficou assim", numeros))

    for(n in numeros){
        if(n %% 2 == 0){
            append(numerosPar, n)
        }else{
            append(numerosImpar, n)
        }
    }

    print(paste("Pares:", numerosPar))
    print(paste("Ímpares:", numerosImpar))
    print(paste("Soma pares:", sum(numerosPar)))
    print(paste("Soma ímpares:", sum(numerosImpar)))
    print(paste("Média:", sum(numeros)/quantidade))
}

trataValores()